### DISCUSSION TITLE
Is rabbit_auth_backend_cache supported to avoid too many connections to ldap in chef-cookbook syntax?

### DISCUSSION THREAD

**Discussion by kmishra-BRONZA** [rabbitmq/chef-cookbook#581] [General]
We have a requirement to cache ldap connections in rabbitmq server. We have tried many ways to place below code snippet in chef json format but cache related config is not getting enabled in node when we run chef-client.

First we have enabled **rabbitmq_auth_backend_cache** plugin in enabled_plugins file

`{
  "name": "rabbitmq-ldap-role",
  "description": "some description",
  "json_class": "Chef::Role",
  "default_attributes": {},
  "override_attributes": {
    "rabbitmq": {
      "auth_backends": "rabbit_auth_backend_cache,rabbit_auth_backend_ldap,rabbit_auth_backend_internal",
      "ldap": {
        //few ldap configurations
      },
      "auth": {
        "cache": {
          "enabled": true,
          "conf": {
            "cached_backend": "rabbit_auth_backend_ldap",
            "cache_ttl": 300000
          }
        }
      }
    }
  },
  "chef_type": "role",
  "run_list": []
}`

Could you please help me on it?

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
The only option for less popular plugins is to use an `advanced.config` (or arbitrary `rabbitmq.config` lines, in case of this cookbook).

**Comment by michaelklishin** [comment]
[These lines](https://github.com/rabbitmq/chef-cookbook/blob/v5.x/templates/default/rabbitmq.config.erb#L148) in the `rabbitmq.config` template seem to try to make exactly that possible.
